src/arrayMethodSort.js

```diff
@@ -4,8 +4,59 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const length = this.length;
+
+    let comparator;
+
+    if (typeof compareFunction !== 'function') {
+      comparator = function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA < strB) {
+          return -1;
+        }
+
+        if (strA > strB) {
+          return 1;
+        }
+
+        return 0;
+      };
+    } else {
+      comparator = compareFunction;
+    }
+
+    function quickSort(arr, left, right) {
+      if (left < right) {
+        const pivotIndex = partition(arr, left, right);
+
+        quickSort(arr, left, pivotIndex - 1);
+
+        quickSort(arr, pivotIndex + 1, right);
+      }
+    }
+
+    function partition(arr, left, right) {
+      const pivot = arr[right];
+      let i = left - 1;
+
+      for (let j = left; j < right; j++) {
+        if (comparator(arr[j], pivot) <= 0) {
+          i++;
+          [arr[i], arr[j]] = [arr[j], arr[i]];
+        }
```

Comment:
Good use of array destructuring for swapping elements. This is a clean and concise way to swap elements without the need for a temporary variable.